flexget/plugins/operate/free_space.py

```diff
@@ -23,7 +23,7 @@ def get_free_space(config, task):
                 config['host'],
                 config['port'],
                 config['user'],
-                config['ssh_key_filepath'],
+                key_filename=config['ssh_key_filepath'],
```

Comment:
I would not recommend this change. We would want to use the SSH Clients options as the key name here, for easier reading the point could be made to update the config key via the [prepare_key ](https://github.com/Flexget/Flexget/blob/develop/flexget/plugins/operate/free_space.py#L93) This would allow you to do something like
```
if config.get('ssh_key_filepath'):
    config['key_path'] = config.pop('ssh_key_file'
```

But we could always use the SSH clients params as the keys and explain their usage in the wiki.

Reply:
Thanks for your suggestion. I set it according to the official configuration, but at least in the latest paramiko module (2.9.2), if the ssh.connect function does not use keyword arguments, then the key pair(which set in the config file) will be passed as the password as the fourth parameter Into the function, the remaining space of the remote server cannot be obtained.
My way of writing may not be perfect, I didn't take into account the situation of no configuration, but now the current code can't work directly, because it uses the key pair in the parameter as the password to connect to the remote server.

Reply:
Yeah the way it is being built it's using positional indexes. So there are 2 options, we build the connect string using defaults in a key=value pairing. Or you just add empty strings to positional arguments that we don't use anyway.
Below works by the way, 1st empty string is password field, 2nd is pkey field.
```
            ssh.connect(
                config['host'],
                config['port'],
                config['user'],
                '',
                '',
                config['ssh_key_filepath'],
                timeout=5000,
'''

I prefer adding the values with defaults to the schema so they can be used with explanations on how they can be used in the wiki myself.

```
                    'path': {'type': 'string'},
                    'port': {'type': 'integer', 'default': 22},
                    'host': {'type': 'string'},
                    'user': {'type': 'string'},
                    'password': {'type': 'string', 'default': ''}, """ password default empty """
                    'pkey': {'type': 'string', 'default': ''}, """ Single private key """
                    'ssh_key_filepath': {'type': 'string'}, """ single/List of rsa, private keys """
                    'allotment': {'type': 'number', 'default': -1},

 """ new ssh.connect() """
            ssh.connect(
                config['host'],
                config['port'],
                config['user'],
                config['password'],
                config['pkey'],
                config['ssh_key_filepath'],
                timeout=5000,

```

Reply:
I don't have much experience in python programming. It was the first time I used flexget that I didn't get the results I expected, so I had to choose to read the source code to solve my confusion. Thanks for your suggestion, I learned a lot.
I've made the changes as you suggested, default value of ssh.connect parameter is None instead of empty stringand. And sorry I don't know what allotment is, I didn't find it in the parametersbut.I also don't know how to submit changes to the wiki, but I'd love to be a contributor to a piece of software I love, thanks.